go/base/static/js/test/apps/dialogue/testHelpers.js

```diff
@@ -6,96 +6,118 @@
       DialogueModel = dialogue.models.DialogueModel,
       DialogueDiagramView = dialogue.diagram.DialogueDiagramView;
 
-  var modelData = {
-    campaign_id: 'campaign-1',
-    conversation_key: 'conversation-1',
-    channel_types: [],
-    poll_metadata: {
-      repeatable: false,
-      delivery_class: 'ussd'
-    },
-    start_state: {uuid: 'state1'},
-    groups: [{
-      key: 'group1',
-      name: 'Group 1',
-      urls: null
-    }, {
-      key: 'group2',
-      name: 'Group 2',
-      urls: null
-    }, {
-      key: 'group3',
-      name: 'Group 3',
-      urls: null
-    }],
-    urls: {
-      show: 'conversation:show:conversation-1'
-    },
-    states: [{
-      uuid: 'state1',
-      name: 'Message 1',
-      type: 'choice',
-      store_as: 'message-1',
-      text: 'What is your favourite colour?',
-      user_defined_store_as: false,
-      entry_endpoint: {uuid: 'endpoint0'},
-      choice_endpoints: [{
-        value: 'red',
-        label: 'Red',
-        uuid: 'endpoint1',
-        user_defined_value: false
-      }, {
-        value: 'blue',
-        label: 'Blue',
-        uuid: 'endpoint2',
-        user_defined_value : false
-      }],
-    }, {
-      uuid: 'state2',
-      name: 'Message 2',
-      type: 'freetext',
-      store_as: 'message-2',
-      user_defined_store_as: false,
-      entry_endpoint: {uuid: 'endpoint3'},
-      exit_endpoint: {uuid: 'endpoint4'},
-      text: 'What is your name?'
-    }, {
-      uuid: 'state3',
-      name: 'Ending 1',
-      type: 'end',
-      store_as: 'ending-1',
-      user_defined_store_as: false,
-      entry_endpoint: {uuid: 'endpoint5'},
-      text: 'Thank you for taking our survey'
-    }, {
-      uuid: 'state4',
-      name: 'Dummy Message 1',
-      type: 'dummy',
-      user_defined_store_as: false,
-      store_as: 'dummy-message-1',
-      entry_endpoint: {uuid: 'endpoint6'},
-      exit_endpoint: {uuid: 'endpoint7'}
-    }, {
-      uuid: 'state5',
-      name: 'Message 5',
-      type: 'group',
-      group: {key: 'group1'},
-      user_defined_store_as: false,
-      store_as: 'message-5',
-      entry_endpoint: {uuid: 'endpoint8'},
-      exit_endpoint: {uuid: 'endpoint9'}
-    }],
-    connections: [{
-     uuid: 'endpoint1-endpoint3',
-     source: {uuid: 'endpoint1'},
-     target: {uuid: 'endpoint3'}
-    }]
+   var modelData = function() {
+     return {
```

Comment:
Sure :) 
Each time we were making changes to the models, we'd have irrelevant test failures, since the tests were basically expecting the models to look like some hardcoded data. To prevent these irrelevant test failures, I instead changed to comparing with a *new* copy of the model data in those tests, so that those tests can only care about the changes relevant to the test. This was related to this PR because of the model changes we made in this PR, though thinking about it now, it probably should've been a separate PR :/

So, that's the reasoning behind trying to get a *new* copy of the data. Having a function means we can ensure a new copy of the model data just by calling the function, as opposed to taking an existing data structure and trying to deep copy it. Deep copying in js is complicated, error prone, and not available in underscorejs ([*because* it is complicated and error prone](https://github.com/jashkenas/underscore/issues/162)).